File: client/src/app/_models/message.ts
import { Gender } from "../_enums/gender.enum";

export interface Message {
	id: number;
	senderId: number;
	senderUsername: string;
	senderPhotoUrl: string;
	senderGender: Gender;
	receiverId: number;
	receiverUsername: string;
	receiverPhotoUrl: string;
	receiverGender: Gender;
	content: string;
	dateRead?: Date;
	messageSent: Date;
}

File: API/Entities/Message.cs
using System;

namespace API.Entities
{
    public class Message
    {
        public int Id { get; set; }
        public int SenderId { get; set; }
        public string SenderUsername { get; set; }
        public AppUser Sender { get; set; }
        public int ReceiverId { get; set; }
        public string ReceiverUsername { get; set; }
        public AppUser Receiver { get; set; }
        public string Content { get; set; }
        public DateTime? DateRead { get; set; }
		public DateTime MessageSent { get; set; } = DateTime.UtcNow;
        public bool SenderDeleted { get; set; }
        public bool ReceiverDeleted { get; set; }
	}
}